diff --git a/src/server/game/Cache/WhoListCacheMgr.cpp b/src/server/game/Cache/WhoListCacheMgr.cpp
index 1c5bbefe8..3f6b11d58 100644
--- a/src/server/game/Cache/WhoListCacheMgr.cpp
+++ b/src/server/game/Cache/WhoListCacheMgr.cpp
@@ -58,6 +58,6 @@ void WhoListCacheMgr::Update()
         _whoListStorage.emplace_back(player->GetGUID(), player->GetTeamId(), player->GetSession()->GetSecurity(), player->GetLevel(),
             player->getClass(), player->getRace(),
             (player->IsSpectator() ? AREA_DALARAN : player->GetZoneId()), player->getGender(), player->IsVisible(),
-            widePlayerName, wideGuildName, playerName, guildName);
+            widePlayerName, wideGuildName, playerName, guildName, player->GetSession()->IsBot());
     }
 }
diff --git a/src/server/game/Cache/WhoListCacheMgr.h b/src/server/game/Cache/WhoListCacheMgr.h
index cfa7f08ab..0321e81ec 100644
--- a/src/server/game/Cache/WhoListCacheMgr.h
+++ b/src/server/game/Cache/WhoListCacheMgr.h
@@ -26,7 +26,7 @@ class WhoListPlayerInfo
 {
 public:
     WhoListPlayerInfo(ObjectGuid guid, TeamId team, AccountTypes security, uint8 level, uint8 clss, uint8 race, uint32 zoneid, uint8 gender, bool visible, std::wstring const& widePlayerName,
-        std::wstring const& wideGuildName, std::string const& playerName, std::string const& guildName) :
+        std::wstring const& wideGuildName, std::string const& playerName, std::string const& guildName, bool isBot) :
         _guid(guid),
         _team(team),
         _security(security),
@@ -39,7 +39,8 @@ public:
         _widePlayerName(widePlayerName),
         _wideGuildName(wideGuildName),
         _playerName(playerName),
-        _guildName(guildName) { }
+        _guildName(guildName),
+        _isBot(isBot) { }
 
     ObjectGuid GetGuid() const { return _guid; }
     TeamId GetTeamId() const { return _team; }
@@ -54,6 +55,7 @@ public:
     std::wstring const& GetWideGuildName() const { return _wideGuildName; }
     std::string const& GetPlayerName() const { return _playerName; }
     std::string const& GetGuildName() const { return _guildName; }
+    bool IsBot() const { return _isBot; }
 
 private:
     ObjectGuid _guid;
@@ -69,6 +71,7 @@ private:
     std::wstring _wideGuildName;
     std::string _playerName;
     std::string _guildName;
+    bool _isBot;
 };
 
 using WhoListInfoVector = std::vector<WhoListPlayerInfo>;
diff --git a/src/server/game/Handlers/MiscHandler.cpp b/src/server/game/Handlers/MiscHandler.cpp
index 4c3ba5502..a5e7d7528 100644
--- a/src/server/game/Handlers/MiscHandler.cpp
+++ b/src/server/game/Handlers/MiscHandler.cpp
@@ -310,6 +310,11 @@ void WorldSession::HandleWhoOpcode(WorldPacket& recvData)
             continue;
         }
 
+        if( target.IsBot() )
+        {
+            continue;
+        }
+
         // check if target's level is in level range
         uint8 lvl = target.GetLevel();
         if (lvl < levelMin || lvl > levelMax)
